icicle/src/merkle-tree/mmcs.cu

```diff
@@ -0,0 +1,458 @@
+#include "hash/hash.cuh"
+#include "merkle-tree/merkle.cuh"
+#include "matrix/matrix.cuh"
+
+#include <algorithm>
+
+using matrix::Matrix;
+
+namespace merkle_tree {
+
+  template <typename L, typename D>
+  cudaError_t hash_leaves(
+    const Matrix<L>* leaves,
+    unsigned int number_of_inputs,
+    uint64_t number_of_rows,
```

Comment:
what is number_of_rows? of what? Matrices have [width, height] fields so it's not clear to me what this refers to.

Reply:
Number of rows is the number of rows of the matrices. This is equal to their `height`